### PR TITLE
Support Network Creation with PoolUUID

### PR DESCRIPTION
**Pull request by sriram-gn**
It's not possible to create a network without PoolUUID as of now, hence, we have to support creation of network with "PoolUUID" which would mean reversal of reconciliation order from "network first" approach to "addresspool first" approach.

It's important to note that UUID of first addresspool in the list of "AssociatedAddressPools" will be used during creation of network and this will become the primary IP pool of the network.

Test Cases:
1. PASS - Verify storage network and addresspools are created successfully as per the specs applied.
2. PASS - Verify network of type "other" and associated addresspools are created successfully as per the specs applied.

### CODE REVIEW

**Comment by yjian118** [comment]
"It's important to note that UUID of first addresspool in the list of "AssociatedAddressPools" will be used during creation of network and this will become the primary IP pool of the network."

Can we make it as a comment somewhere?

File: controllers/host/platformnetworks.go [hunk 1]
```diff
@@ -184,6 +184,11 @@ func (r *HostReconciler) IsNetworkUpdateRequired(network_instance *starlingxv1.P
 
 	if system_network != nil {
 		uuid = system_network.UUID
+	} else {
```
Comment (yjian118) [inline controllers/host/platformnetworks.go:187]: if system_network == nil {
    // Network is not configured in the system, assign the pool and create the network  
     new section
}

Comment (yjian118) [reply controllers/host/platformnetworks.go:187]: BTW, it is confusing for me to think about what is "system_network", can you rename it as "current_network"?

Comment (sriram-gn) [reply controllers/host/platformnetworks.go]: > if system_network == nil {
> // Network is not configured in the system, assign the pool and create the network
> new section
> }
Done.

> BTW, it is confusing for me to think about what is "system_network", can you rename it as "current_network"?
Done.

File: controllers/host/platformnetworks.go [hunk 2]
```diff
@@ -184,6 +184,11 @@ func (r *HostReconciler) IsNetworkUpdateRequired(network_instance *starlingxv1.P
...
+		// This is required while creating the network only and
+		// internally managed by the system during reconfiguration.
+		opts.PoolUUID = &primary_address_pool.ID
+		delta.WriteString(fmt.Sprintf("\t+PoolUUID: %v\n", *opts.PoolUUID))
```
[the remark this answers is not in the document]
Comment (sriram-gn) [reply controllers/host/platformnetworks.go:189]: Done.

**Comment by sriram-gn** [comment]
> "It's important to note that UUID of first addresspool in the list of "AssociatedAddressPools" will be used during creation of network and this will become the primary IP pool of the network."
> 
> Can we make it as a comment somewhere?

Done, it's here:
https://github.com/Wind-River/cloud-platform-deployment-manager/pull/370/files#diff-5067c7a106f9179836e1f03276e0cdd0b67a3c437cec1ce2b0a80aed7986fbb7R881

**Review by yjian118 — APPROVED** [review]

**Pull request merged**